applications/GeoMechanicsApplication/custom_processes/geo_apply_constant_scalar_value_process.cpp

```diff
@@ -8,219 +8,81 @@
 //  License:         geo_mechanics_application/license.txt
 //
 //  Main authors:    Riccardo Rossi
+//                   Richard Faasse
 //
 
 #include "geo_apply_constant_scalar_value_process.h"
-#include "containers/model.h"
+#include "containers/variable.h"
+#include "containers/variable_data.h"
+#include "includes/kratos_components.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
 #include "utilities/variable_utils.h"
 
 namespace Kratos
 {
 
-KRATOS_CREATE_LOCAL_FLAG(GeoApplyConstantScalarValueProcess, VARIABLE_IS_FIXED, 0)
-
-GeoApplyConstantScalarValueProcess::GeoApplyConstantScalarValueProcess(Model& rModel, Parameters ThisParameters)
-    : GeoApplyConstantScalarValueProcess(
-          rModel.GetModelPart(ThisParameters["model_part_name"].GetString()), ThisParameters)
-{
-}
-
-GeoApplyConstantScalarValueProcess::GeoApplyConstantScalarValueProcess(ModelPart& rModelPart, Parameters ThisParameters)
-    : Process(Flags()), mrModelPart(rModelPart)
+GeoApplyConstantScalarValueProcess::GeoApplyConstantScalarValueProcess(ModelPart& rModelPart,
+                                                                       const Parameters& ThisParameters)
+    : mrModelPart(rModelPart)
 {
-    KRATOS_TRY
-
-    // Some values need to be mandatorily prescribed since no meaningful default value exist.
-    // So that an error is thrown if they don't exist
     KRATOS_ERROR_IF_NOT(ThisParameters.Has("value"))
-        << "Missing 'value' parameter in ThisParameters" << std::endl;
+        << "Missing 'value' parameter in the parameters of '"
+        << GeoApplyConstantScalarValueProcess::Info() << "'" << std::endl;
     KRATOS_ERROR_IF_NOT(ThisParameters.Has("variable_name"))
-        << "Missing 'variable_name' parameter in ThisParameters" << std::endl;
-    KRATOS_ERROR_IF_NOT(ThisParameters.Has("model_part_name"))
-        << "Missing 'model_part_name' parameter in ThisParameters" << std::endl;
-
-    // Now validate against defaults -- this also ensures no type mismatch
-    ThisParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+        << "Missing 'variable_name' parameter in the parameters of '"
+        << GeoApplyConstantScalarValueProcess::Info() << "'" << std::endl;
 
     mVariableName = ThisParameters["variable_name"].GetString();
-    this->Set(VARIABLE_IS_FIXED, ThisParameters["is_fixed"].GetBool());
+    mIsFixed      = ThisParameters.Has("is_fixed") ? ThisParameters["is_fixed"].GetBool() : false;
```

Comment:
Does this change mean that parameter `"is_fixed"` is no longer required, but optional (having a default value of `false` if not provided)? If yes, could you please help me understand why this change is needed?

Also, if this functional change is needed, we are repeating the default value when `"is_fixed"` is not provided (the default value of `mIsFixed` is also set in the header file, which is great). So perhaps we can rewrite this statement to:
```suggestion
    if (ThisParameters.Has("is_fixed")) mIsFixed = ThisParameters["is_fixed"].GetBool();
```

Reply:
As discussed, it was handled by using the default parameters before, so the behavior hasn't changed. I implemented your suggestion, because indeed the default is already there in the header.